tket/src/Placement/include/Placement/NeighbourPlacements.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <map>
+
+#include "Placement.hpp"
+#include "TokenSwapping/SwapFunctions.hpp"
+#include "Utils/BiMapHeaders.hpp"
+#include "Utils/RNG.hpp"
+
+namespace tket {
+
+/**
+ * @brief Given a placement map generates `n` nearby placement maps.
+ *
+ * Based on an architecture and a placement map, generates random
+ * placements that can be achieved with `m` swaps.
+ *
+ * Optionally uses token swapping optimisations to try to ensure
+ * that the generated placements cannot be obtained in less than `m`
+ * swaps, but this cannot be guaranteed.
+ */
+class NeighbourPlacements {
+ public:
+  using SwapVec = std::vector<Swap>;
+  using NodeSwap = std::pair<Node, Node>;
+  using NodeSwapVec = std::vector<NodeSwap>;
+  struct Result {
```

Comment:
Would it be possible to add here the actual dist of the result from the initial mapping?

Reply:
The optimal dist is very hard to compute. I can add a `dist()` function that returns `swaps.size()`. Would that be useful?

Reply:
I think a swaps.size() function looks like the best solution for this.

Reply:
Just to avoid confusion: `Result.swaps` is a vector, so it already has a `.size()`; are you fine with me not adding anythign?

Reply:
Yes that is fine